### DISCUSSION TITLE
修改的背景图片（侧边栏以及明暗背景）能够在本地正常显示，但是部署之后就无法显示了

### DISCUSSION THREAD

**Discussion by lingxin12** [YunYouJun/hexo-theme-starry#111] [Q&A 疑难杂症] [answered]
作者你好。

我在修改yun.yml配置文件的过程中，修改了明暗背景图片以及侧边栏的背景，然后生成静态文件后能够正常在本地显示，可是部署成功后再访问就无法显示了。（背景和网站logo都改不了，只有头像正常显示）

请问是因为图片分辨率对不上的原因还是因为什么？

下面是我的配置
[yun.txt](https://github.com/YunYouJun/hexo-theme-yun/files/6090447/yun.txt)
（由于不知道github的编辑代码块怎么用就转成txt了）

**Comment by YunYouJun** [comment]
检查网站路径是否正确。检查本地是否已正确生成对应静态文件。

既然如此，问题只可能出在部署方面。最好提供一下预览链接。

**Reply by lingxin12** [reply]
这个博客的链接：lingxinblog.cn

这是yun.yml中相关背景图片的配置：
![AHUPR3VQ1V{U{I8M33F MVH](https://user-images.githubusercontent.com/57349676/110119016-29b3ad00-7df6-11eb-9fe7-70a7d57a4729.png)
![@L6GP7F6IU LRJ}~1RC){_2](https://user-images.githubusercontent.com/57349676/110119017-29b3ad00-7df6-11eb-8e17-6245841e1074.png)
![9`}`6~%9(UC@7 @QG}}D}TS](https://user-images.githubusercontent.com/57349676/110119015-29b3ad00-7df6-11eb-991d-9f50715dee9d.png)

图片均放在了themes\yun\source\images文件夹下
这是最近一次生成静态文件和部署的报告
![%0HHT5LC D74OXDW%%L@XU8](https://user-images.githubusercontent.com/57349676/110119351-a8104f00-7df6-11eb-92b4-fc6865e41ead.png)
![54IQ{WSHIU5E~)_UGMJ9Q J](https://user-images.githubusercontent.com/57349676/110119365-acd50300-7df6-11eb-86e4-44b17b7ed129.png)
![$9BO8D`NN SQ Y3W7( W}$Y](https://user-images.githubusercontent.com/57349676/110119377-b199b700-7df6-11eb-9441-c10db0c7a469.png)
![CDO~9U9N`KBEXB`G~$4N5_W](https://user-images.githubusercontent.com/57349676/110119389-b494a780-7df6-11eb-896d-f0c0de9179ce.png)

**Reply by YunYouJun** [reply]
> 图片均放在了themes\yun\source\images文件夹下

你应当放在你自己的 hexo 目录下，而非主题的目录下。主题是一个额外的依赖，最好不要对其任何内容进行修改。

**Comment by lingxin12** [comment, marked as answer]
作者大大我傻了。（刚才一顿我也不知道操作就搞好了）

刚刚的历程是这样的：
==================================
本来我已经把主题文件增加的东西都删除了，然后把所有自己的文件都转移到了hexo的source文件夹中。重新生成静态文件，发现还是同一个问题，本地正常显示，但部署后无法加载背景图片（背景主题色也无法加载）。

于是我想重头再来，把_data/yun.yml备份后删除，music和images文件都保留了下来。然后在cmd中将yun主题文件夹强制删除。然后重新下载yun主题，决定重新配置。

下载完主题后，在hexo目录的source下新建了_data/yun.yml（里面是空的），然后在yun主题中git pull，再然后重新生成静态文件并本地预览，按照我的想法页面应该是默认的配置，就是作者大大的预览主页那样。但是我第一个疑问就来了，我发现本地预览后，和我删除主题之前的配置一摸一样！！！！！头像、音乐、背景、主题色、导航等一样没有变。（这里我非常好奇为什么会这样？？）

然后我就猜测主题是我强制删除的，可能还有上一次配置残留，后来我就把hexo下的public以及.deploy_git都删除了，把生成的静态文件和github库的内容都删完，然后重新清缓存生成静态文件，然后本地预览，发现还是没有变化！！！！（此时yun.yml中还是一片空白），然后部署后访问，突然发现图片和主题色都正常加载了。（虽然有点慢2333）

此时我满脸黑人问号？？？？？这到底怎么回事？

我有点害怕，如果以后我还继续修改主题配置，把备份的yun.yml和新写的配置重新写入新建的yun.yml中会不会又回到之前痛苦的循环之中，作者大大可不可以给点建议，谢谢了！

**Reply by YunYouJun** [reply]
是不是没清缓存，使用 ctrl f5 强制刷新。图片资源，浏览器会默认缓存。
其次用到什么配置，就把什么写到 yun.yml 里就好了。（这时就可以任意删除/更新主题，而不担心丢失配置。）

**Reply by lingxin12** [reply]
目前已解决，实在感谢作者大大的解答!!!!（磕头）
